### PR TITLE
Single Event Page!

### PR DESCRIPTION
**Pull request by esmail-elmoussel**
I've added 3 packages.. 
1) react-responsive-modal: to add a nice popup modal. 
2) react-dynamic-form-builder: to help with dynamic form implementation.
3) core-js: to fix previous package issues.

### CODE REVIEW

File: src/modules/events/components/singleEventPage/components/EventDetails/EventDetails.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import ApplyButton from '../ApplyButton/ApplyButton';
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import {
+    faMapMarkerAlt,
+    faClock
+} from "@fortawesome/free-solid-svg-icons";
+import styles from './style.module.css';
+
+
+class Gallery extends Component {
```
Comment (dvmhmdsd) [inline src/modules/events/components/singleEventPage/components/EventDetails/EventDetails.js]: Is this component `Gallery` or `EventDetails`

File: src/modules/events/components/singleEventPage/components/ApplyButton/style.module.css [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+.apply-btn:focus {
+    background-color: var(--btn-hover);
+    box-shadow: 0 12px 16px 0 rgba(0,0,0,0.24),0 17px 50px 0 rgba(0,0,0,0.19);
+}
+
+.apply-btn:focus {
+    transform: translateY(10px);
+}
+
+@keyframes :local(customEnterAnimation) {
```
Comment (dvmhmdsd) [inline src/modules/events/components/singleEventPage/components/ApplyButton/style.module.css]: I get an error here from `eslint` 

`Identifier expected`

File: src/modules/events/components/singleEventPage/components/Gallery/Gallery.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    items: 1,
+  },
+};
+
+class Gallery extends Component {
+  render() {
+    const { images } = this.props
+    return (
+      <section className='my-5'>
+        <h3 className='h3 mt-5 mb-4' style={{ color: "var(--text-secondary)" }}>Event Gallery</h3>
```
Comment (dvmhmdsd) [inline src/modules/events/components/singleEventPage/components/Gallery/Gallery.js]: This should be `h2`

File: src/shared/Input.js [hunk 4]
```diff
@@ -275,13 +265,13 @@ export default class Input extends Component {
   };
 
   addClass = (e) => {
-    if (this.props.className) {
-      e.target.classList.add(this.props.className);
+    if (this.props.classes) {
+      e.target.classList.add(this.props.classes);
```
Comment (dvmhmdsd) [inline src/shared/Input.js]: It should be `className` to get the default `className` of the element. And if didn't use it, remove the file.

**Review by dvmhmdsd — CHANGES_REQUESTED** [review]
- ` StartTime` is optional, check `/events/5`

- The pullets of the gallery carousel is not accessible.

**Comment by dvmhmdsd** [comment]
It should show `Not Found` if the id in the `url` is not in the events data.

**Comment by esmail-elmoussel** [comment]
> It should show `Not Found` if the id in the `url` is not in the events data.

I think we should build a nice 404 NOT FOUND component.

**Comment by dvmhmdsd** [comment]
> > It should show `Not Found` if the id in the `url` is not in the events data.
> 
> I think we should build a nice 404 NOT FOUND component.

We'll, as in the `TODOs` file. But make it for now like the `singleServicePage`.

**Review by dvmhmdsd — CHANGES_REQUESTED** [review]
1. Make the components independant. (Apply button shouldn't have the `cover & `form`).

**Pull request merged**